src/CachePlugin.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Allow for caching a response.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class CachePlugin implements Plugin
+{
+    /**
+     * @var CacheItemPoolInterface
+     */
+    private $pool;
+
+    /**
+     * Default time to store object in cache. This value is used if CachePlugin::respectCacheHeaders is false or
+     * if cache headers are missing.
+     *
+     * @var int
+     */
+    private $defaultTtl;
+
+    /**
+     * Look at the cache headers to know how long this response is going to be cached.
+     *
+     * @var bool
+     */
+    private $respectCacheHeaders;
+
+    /**
+     * @param CacheItemPoolInterface $pool
+     * @param array                  $options
+     */
+    public function __construct(CacheItemPoolInterface $pool, array $options = array())
+    {
```

Comment:
we depend on the symfony OptionsResolver already. i would propose to emply the resolver here to do validation.

the plus side of the array is that adding more optional configuration with default value will be simply with an array, and much trickier if they all need an explicit order. (for BC, parameter order can not change, so required parameters could end up after optional parameters and whatnot.